docs/source/essentials/queries.md

```diff
@@ -0,0 +1,280 @@
+
+This page assumes some familiarity with building GraphQL queries. If you'd like a refresher, we recommend [reading this guide](http://graphql.org/learn/queries/) and practicing [running queries in GraphiQL](https://graphql.github.io/swapi-graphql/).
+Since Apollo Client queries are just standard GraphQL, anything you can type into the GraphiQL query explorer can also be put into `.graphql` files in your project.
+
+The following examples assume that you've already set up Apollo Client for your Android/Java application. Read our [getting started](./get-started.html) guide if you need help with either of those steps.
+
+> All code snippets are taken from the apollo-sample project and can be found [here](https://github.com/apollographql/apollo-android/tree/master/apollo-sample).
+
+Apollo-android takes a schema and a set of `.graphql` files and uses these to generate code you can use to execute queries and access typed results.
+
+> All `.graphql` files in your project (or the subset you specify as input to `apollo-codegen` if you customize the script you define as the code generation build phase) will be combined and treated as one big GraphQL document. That means fragments defined in one `.graphql` file are available to all other `.graphql` files for example, but it also means operation names and fragment names have to be unique and you will receive validation errors if they are not.
+
+<h2 id="creating-queries">Creating queries</h2>
+
+Queries are represented as instances of generated classes conforming to the `GraphQLQuery` protocol. Constructor arguments can be used to define query variables if needed.
+You pass a query object to `ApolloClient#query(query)` to send the query to the server, execute it, and receive results.
+
+For example, if you define a query called `FeedQuery`:
+
+```graphql
+query FeedQuery($type: FeedType!, $limit: Int!) {
+  feedEntries: feed(type: $type, limit: $limit) {
+    id
+    repository {
+      name
+      full_name
+      owner {
+        login
+      }
+    }
+    postedBy {
+      login
+    }
+  }
+}
+```
+
+Here, `query` is the operation type and `FeedQuery` is the operation name.
+Apollo-android will generate a `FeedQuery` class that you can construct (with variables) and pass to `ApolloClient#query(query)`:
+
+```java
+apolloClient().query(feedQuery)
+        .enqueue(new ApolloCallback<>(new ApolloCall.Callback<FeedQuery.Data>() {
+          @Override public void onResponse(@NotNull Response<FeedQuery.Data> response) {
+            Log.i(TAG, response.toString());
+          }
+
+          @Override public void onFailure(@NotNull ApolloException e) {
+            Log.e(TAG, e.getMessage(), e);
+          }
+        }, uiHandler));
+```
+
+> By default, Apollo will deliver query results on a background, so you have to provide a handler in enqueue if you're using the result to update the UI.
```

Comment:
Maybe add a note about the android support which has a handler you can use to deliver on Main thread

Reply:
On a side note, the `wrap` method(line 34) in ApolloCallback.java under apollo-android-support is used just for tests. Can it be marked `@VisibleForTesting` if it is not supposed to be used as a public API?